anomalib/models/draem/loss.py

```diff
@@ -0,0 +1,204 @@
+"""Loss functions for DRAEM implementation."""
+
+# Original Code
+# Copyright (c) 2022 VitjanZ
+# https://github.com/VitjanZ/DRAEM.
+# SPDX-License-Identifier: MIT
+#
+# Modified
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=invalid-name
+
+from math import exp
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch import Tensor, nn
+
+
+class FocalLoss(nn.Module):
```

Comment:
Have you experimented with torchvision's focal loss? While the implementation looks different, the linked repo and torchvision use it for RetinaNet. So replacing it might work. Not completely sure though.